google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/test/java/com/google/cloud/spanner/jdbc/ITAbstractJdbcTest.java

```diff
@@ -0,0 +1,205 @@
+      url.append(";credentials=").append(getKeyFile());
+    }
+    appendConnectionUri(url);
+
+    return DriverManager.getConnection(url.toString()).unwrap(CloudSpannerJdbcConnection.class);
+  }
+
+  protected void appendConnectionUri(StringBuilder uri) {}
+
+  /**
+   * Override this method to instruct the test to create a default test table in the form:
+   *
+   * <pre>
+   * CREATE TABLE TEST (ID INT64 NOT NULL, NAME STRING(100) NOT NULL) PRIMARY KEY (ID)
+   * </pre>
+   *
+   * Note that the table is not re-created for each test case, but is preserved between test cases.
+   * It is the responsibility of the test class to either empty the table at the end of each test
+   * case, or keep track of the state of the test table and execute the test cases in a specific
+   * order.
+   *
+   * @return true if the default test table should be created.
+   */
+  protected boolean doCreateDefaultTestTable() {
+    return false;
+  }
+
+  protected boolean doCreateMusicTables() {
+    return false;
+  }
+
+  @Before
+  public void createTestTable() throws SQLException {
+    if (doCreateDefaultTestTable()) {
+      try (Connection connection = createConnection()) {
+        connection.setAutoCommit(true);
+        if (!tableExists(connection, "TEST")) {
+          connection.setAutoCommit(false);
+          connection.createStatement().execute("START BATCH DDL");
+          connection.createStatement().execute(
+              "CREATE TABLE TEST (ID INT64 NOT NULL, NAME STRING(100) NOT NULL) PRIMARY KEY (ID)");
+          connection.createStatement().execute("RUN BATCH");
+        }
+      }
+    }
+  }
+
+  @Before
+  public void createMusicTables() throws SQLException {
+    if (doCreateMusicTables()) {
+      try (Connection connection = createConnection()) {
+        connection.setAutoCommit(true);
+        if (!tableExists(connection, "Singers")) {
+          for (String statement : readStatementsFromFile("CreateMusicTables.sql")) {
+            connection.createStatement().execute(statement);
+          }
+        }
+      }
+    }
+  }
+
+  protected boolean tableExists(Connection connection, String table) throws SQLException {
+    try (ResultSet rs = connection.getMetaData().getTables("", "", table, null)) {
+      if (rs.next()) {
+        if (rs.getString("TABLE_NAME").equalsIgnoreCase(table)) {
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+  protected boolean indexExists(Connection connection, String table, String index)
+      throws SQLException {
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(index));
+    try (PreparedStatement ps = connection.prepareStatement(
+        "SELECT INDEX_NAME FROM INFORMATION_SCHEMA.INDEXES WHERE UPPER(TABLE_NAME)=? AND UPPER(INDEX_NAME)=?")) {
+      ps.setString(1, table);
+      ps.setString(2, index);
+      try (ResultSet rs = ps.executeQuery()) {
+        return rs.next();
+      }
+    }
+  }
+
+  List<String> readStatementsFromFile(String filename) {
```

Comment:
I think this is somewhere else in the tests. Can we use a Util class?

Reply:
You're right, it was already in the `AbstractSqlScriptVerifier` class. I've changed the tests to use that method.